components/meter/meter-linear.js

```diff
@@ -76,14 +77,19 @@ class MeterLinear extends LitElement  {
 		const percentage = this.max > 0 ? this.value / this.max * 100 : 0;
 		const primary = this.percent ? `${Math.floor(percentage)}%` : `${this.value}/${this.max}`;
 		const secondaryText = this._formatContext(this.text, this.value, this.max);
-		const secondaryTextElement = secondaryText ? html`<div class="d2l-meter-linear-secondary">${secondaryText}</div>` : html``;
+		const secondaryTextElement = secondaryText ? html`<div id="d2l-meter-linear-secondary" class="d2l-meter-linear-secondary">${secondaryText}</div>` : html``;
 		return html `
-			<div role="img" aria-roledescription="progress indicator" aria-label="${secondaryText}: ${primary}">
+			<div
+				role="progressBar"
+				aria-valuenow="${this.value}"
+				aria-valuemax="${this.max}"
+				aria-valuetext="${ifDefined(secondaryText)}"
+				aria-describedby="d2l-meter-linear-primary d2l-meter-linear-secondary">
 				<div class="d2l-meter-linear-full-bar">
 					<div class="d2l-meter-linear-inner-bar" style="width:${percentage}%;"></div>
 				</div>
 				<div class="d2l-body-small d2l-meter-linear-text">
-					<div class="d2l-meter-linear-primary">${primary}&nbsp;</div>
+					<div id="d2l-meter-linear-primary" class="d2l-meter-linear-primary">${primary}&nbsp;</div>
```

Comment:
In browsers who don't support shadow DOM, these `ID`s aren't going to be unique when there are multiple meters on the page. If this approach works well, it's probably worth porting the [ID generator stuff](https://github.com/Brightspace/d2l-polymer-behaviors-ui/blob/master/d2l-id.js) into core and you can just generate a couple IDs to use.

Reply:
I was wondering about that. Cool, I'll see about doing that.